doc/src/manual/multi-threading.md

```diff
@@ -149,26 +149,37 @@ and nothing promised here can be assumed if you do not observe that
 requirement. The observed results may be highly unintuitive.
 
 The best way to ensure this is to acquire a lock around any access to data that
-can be observed from multiple threads. For example, in most cases you should
-use the following code pattern:
+can be observed from multiple threads. If there is only a single lock, and we require that any thread has to "hold the lock" in order to read or mutate a variable, we ensure that the operations can never happen simultaneously.
+
+For example, we can create a lock `my_lock`, and lock it while we mutate a variable `my_variable`:
 
 ```julia-repl
-julia> lock(lk) do
-           use(a)
-       end
+julia> my_lock = ReentrantLock();
+
+julia> my_variable = [1, 2, 3];
+
+julia> my_operation() = (my_variable[1] = 100);
 
 julia> begin
-           lock(lk)
-           try
-               use(a)
-           finally
-               unlock(lk)
-           end
+           lock(my_lock)
+           my_operation()
```

Comment:
there should always be a try/finally block following `lock`, or use one of the syntax forms (such as `@lock my_lock my_operations()`) that includes it for you

Reply:
Right, which is explained not in the immediate paragraph, but the one after. I find it to be a smoother conceptual journey to start with the simplest example, and build on it. Is the problem/worry that someone might copy that example without reading the context? If yes, would a good fix be to comment "#dont copy this code"?

We are splitting hairs, and I am fine with making all examples "propper". We can then annotate the `finally` clause with ”# ensures that lock is always unlocked", which also provides the very explicit guidance that I am trying to add here.

Reply:
I think most examples should be proper. The macro form might be best here, since it is simplest and also the most common form to use or encounter

Reply:
Alright, so instead of going basic and building up, we are showing the shortest and best first. This is worse for educational purposes, but better for those who want the TL;DR, which I guess is a lot of people when it comes to documentation. I will reorder the examples in the opposite order, and make them all propper.

Reply:
This should be resolved by the latest commit. take a look and see if you agree.